connectivity/suite.go

```diff
@@ -240,6 +254,74 @@ func Run(ctx context.Context, ct *check.ConnectivityTest) error {
 			return check.ResultDrop, check.ResultNone
 		})
 
+	// Tests with deny policy
+	if ct.DenyPolicy() {
+		// This policy denies ingress into client from client2.
+		ct.NewTest("client-ingress-deny").WithPolicy(clientIngressFromClient2DenyPolicyYAML).
+			WithScenarios(
+				tests.ClientToClient(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "client") {
+					return check.ResultOK, check.ResultDrop
+				}
+				return check.ResultOK, check.ResultOK
+			})
+
+		// This policy allows ingress to echo only from client with a label 'other:client'
+		ct.NewTest("echo-ingress-deny").WithPolicy(echoIngressFromOtherClientDenyPolicyYAML).
+			WithScenarios(
+				tests.PodToPod(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "echo") {
+					return check.ResultDrop, check.ResultDrop
+				}
```

Comment:
Based on this echo pods would be dropping traffic regardless the source, while the policy seems to be specific to the "other client"?

Reply:
Also, is the drop happening already at the egress as the return values would seem to indicate. If so how come we get drop also from ingress?